### ISSUE TITLE
Use enabled config to implement cross-database support

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/snowplow#59]
Right now, Snowplow models are encoded as macros in order to implement cross-database support. Instead, we should experiment with making models specifically for each support adapter. I think broadly, this falls into:
- default (snowflake/redshift/postgres)
- bigquery

The models directory might look like:
```
models:
 - common/
 - default/
 - bigquery/
```

Every model would need to have a config like:
```
{{ config(enabled=(var('adapter') == 'default')) }}
or
{{ config(enabled=(var('adapter') == 'bigquery')) }}
```

I don't think we can configure this at a folder level in `dbt_project.yml`, but it might be worth exploring if there's some clever way to implement that.

The results of this experimentation might result is us wanting to _keep_ the adapter macro pattern, but I'm interested in exploring alternative ways of supporting cross-db model implementations in packages!

**Comment by jtcohen6** [comment]
Resolved in #62. Once 0.16.1 comes out, and dbt supports returning boolean YML values from Jinja macros, we can pursue an even-more seamless implementation in `dbt_project.yml`.

**Comment by drewbanin** [comment]
Just an FYI - boolean yaml values will likely be 0.17.0, but regardless this all sounds great!
